src/telemetry/metrics/tls_config_reload.rs

```diff
@@ -31,21 +34,24 @@ enum Status {
 // ===== impl TlsConfigReload =====
 
 impl TlsConfigReload {
-    pub fn success(&mut self, when: &SystemTime) {
-        let t = when
+    pub fn reloaded(&self) {
+        let t = SystemTime::now()
             .duration_since(UNIX_EPOCH)
             .expect("times must be after UNIX epoch")
             .as_secs();
-        self.last_reload = Some(t.into());
 
-        self.by_status.scopes
+        let mut inner = self.0.write().expect("lock");
+        inner.last_reload = Some(t.into());
+
+        inner.by_status.scopes
             .entry(Status::Reloaded)
             .or_insert_with(|| Counter::default())
             .incr()
     }
 
-    pub fn error(&mut self, e: tls::ConfigError) {
-        self.by_status.scopes
+    pub fn failed(&self, e: tls::ConfigError) {
+        let mut inner = self.0.write().expect("lock");
+        inner.by_status.scopes
             .entry(e.into())
             .or_insert_with(|| Counter::default())
```

Comment:
Nit: I know this code wasn't added in this branch, but this could just be `.or_default`.

Reply:
i don't think our version of indexmap has that

Reply:
Nevermind, it looks like `std::collections::HashMap` has an `.or_default()` method but `IndexMap` doesn't.